internal/backend/remote/backend.go

```diff
@@ -716,6 +709,28 @@ func (b *Remote) StateMgr(ctx context.Context, name string) (statemgr.Full, erro
 	}, nil
 }
 
+func localVersionMatchesRemoteVersion(wsv string) error {
+	const latestVersion = "latest"
+
+	if wsv == latestVersion {
+		// We handle the "latest" for compatibility reasons:
+		// TFE and potentially other backends can set the workspace version to "latest"
+		// if they were created using pre-released version of terraform
+		return nil
```

Comment:
**(idea)** We could log a warning here, or fail given that we don't know what did the "latest" mean for the backend when the workspace was created.

Reply:
My preference here is warning if we want to take any action.